### PR TITLE
changefeed: refine the logic about changefeed error

### PR DESCRIPTION
**Pull request by asddongmen**
manual pick this change to:

- [ ] branch 5.1 
- [ ] branch 5.0 
- [ ] branch 4.0

### What problem does this PR solve? 
#2242 

### What is changed and how it works?
- Hiding error message when a changefeed state is normal 
- Improving the logic of func `feedStateManager.handleError()`. 
In the original code, if a fastFail error is not the last item of `errs` , then the fastFail error will not be handle correctly.
 
### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - Breaking backward compatibility

### Release note 

```release-note
None 
```

### CODE REVIEW

**Comment by amyangfei** [comment]
Please add unit tests

**Comment by asddongmen** [comment]
> Please add unit tests

ok~

File: cdc/owner/feed_state_manager.go [hunk 1]
```diff
@@ -257,26 +257,40 @@ func (m *feedStateManager) errorsReportedByProcessors() []*model.RunningError {
...
 
-func (m *feedStateManager) HandleError(errs ...*model.RunningError) {
+func (m *feedStateManager) handleError(errs ...*model.RunningError) {
+	// if one of the error stored by changefeed state(error in the last tick)
+	// is a fast-fail error, the changefeed should be failed
+	if m.state.Info.HasFastFailError() {
+		m.shouldBeRunning = false
+		m.patchState(model.StateFailed)
+		return
+	}
```
Comment (amyangfei) [inline cdc/owner/feed_state_manager.go:267]: Can we remove this logic, since `info.Error` is always set in this function, and checked at L272

Comment (asddongmen) [reply cdc/owner/feed_state_manager.go:267]: yeh， good advice.

File: cdc/model/http_model.go [hunk 2]
```diff
@@ -63,6 +64,20 @@ type ChangefeedCommonInfo struct {
 	RunningError   *RunningError `json:"error"`
 }
 
+// MarshalJSON use to marshal ChangefeedCommonInfo
+func (c ChangefeedCommonInfo) MarshalJSON() ([]byte, error) {
+	// alias the original type to prevent recursive call of MarshalJSON
+	type Alias ChangefeedCommonInfo
```
[the remark this answers is not in the document]
Comment (asddongmen) [reply cdc/model/http_model.go:70]: yes

**Review by 3AceShowHand — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Comment by asddongmen** [comment]
i will fix the ci error ASAP.

**Pull request merged**